tests/python/unittest/test_sparse_operator.py

```diff
@@ -101,21 +105,76 @@ def test_elemwise_add_multiple_stages():
     assert_almost_equal(exec_test.outputs[0].asnumpy(), sp_np1 + sp_np2 + ds_np)
     exec_test.backward(out_grads = exec_test.outputs)
     assert_almost_equal(arr_grads[0].asnumpy(), arr_grads[1].asnumpy())
-'''
+
+
 def test_cast_storage():
-    dns_np = np.array([[0, 0], [5, 10], [0, 0], [0, 0], [0, 0]])
-
-    val = np.array([5, 10])
-    idx = np.array([1])
-    b = mx.nd.array(idx, dtype=np.int32)
-    sp_nd = mx.sparse_nd.array(val, [b], 'row_sparse', (5,2))
-    var = mx.symbol.Variable('sp_data', storage_type='row_sparse')
-    # 1 for row_storage type
-    test = mx.symbol.cast_storage(var, storage_type=1)
-    check_symbolic_forward(test, {'sp_data':sp_nd}, [dns_np])
-'''
+    def test_rsp_to_dns(data, row_idx, shape):
+        rsp = mx.sparse_nd.array(values=data, index_list=[row_idx], storage_type='row_sparse', shape=shape)
+        dns_out = mx.nd.cast_storage(rsp, storage_type='default')
+        dns_expected = np.zeros(shape, dtype=default_dtype())
+        for k, v in enumerate(row_idx):
+            dns_expected[v, :] = data[k]
+        assert same(dns_out.asnumpy(), dns_expected)
+
+    def test_dns_to_rsp(dns_in):
+        dns_in = np.array(dns_in)
+        rsp_out = mx.nd.cast_storage(mx.nd.array(dns_in, dtype=default_dtype()), storage_type='row_sparse')
+        ret = mx.nd.cast_storage(rsp_out, storage_type='default')
+        assert same(ret.asnumpy(), dns_in)
+
+    def test_csr_to_dns(data, indptr, col_idx, shape):
+        indptr = np.array(indptr, dtype=np.int32)
+        col_idx = np.array(col_idx, dtype=np.int32)
+        csr = mx.sparse_nd.array(values=data, index_list=[col_idx, indptr], storage_type='csr', shape=shape,
+                                 aux_types=[np.int32, np.int32])
+        dns_out = mx.nd.cast_storage(csr, storage_type='default')
+        dns_expected = np.zeros(shape, dtype=default_dtype())
+        i = 0
+        while i < len(indptr) - 1:
+            j = indptr[i]
+            while j < indptr[i+1]:
+                dns_expected[i, col_idx[j]] = data[j]
+                j = j + 1
+            i = i + 1
+        assert same(dns_out.asnumpy(), dns_expected)
+
+    def test_dns_to_csr(dns_in):
+        dns_in= np.array(dns_in)
+        csr_out = mx.nd.cast_storage(mx.nd.array(dns_in, dtype=default_dtype()), storage_type='csr')
+        ret = mx.nd.cast_storage(csr_out, storage_type='default')
+        assert same(ret.asnumpy(), dns_in)
+
+    test_rsp_to_dns([], [], (10, 3))
+    test_rsp_to_dns([[1, 2], [3, 4], [5, 6], [7, 8]], [2, 4, 5, 7], (10, 2))
+    test_dns_to_rsp([[0, 1, 0], [0, 2, 0], [3, 0, 0], [0, 0, 4], [5, 6, 0], [0, 0, 7]])
+    test_csr_to_dns([], [0, 0, 0, 0, 0], [], (4, 4))
+    test_csr_to_dns([5, 8, 3, 6], [0, 0, 2, 3, 4], [0, 1, 2, 1], (4, 4))
+    test_dns_to_csr([[0, 1, 0], [0, 2, 0], [3, 0, 0], [0, 0, 4], [5, 6, 0], [0, 0, 7]])
+
```

Comment:
I will add a csr randomizer to the function of creating a random sparse ndarray you wrote after you merge it.